### PR TITLE
Handle file not found when looading restarted summary

### LINKED ISSUE #602: ecl_smspec_install_special_keys : Crash when well name is null pointer
The attached summary files causes crash when using the summary.x tool for version 2.4.0
[summary_crash.zip](https://github.com/equinor/libecl/files/3033065/summary_crash.zip)

image

### PR DESCRIPTION
**Pull request by joakim-hove**
**Issue**
Resolves #602

**Approach**
Check return value - which can legitematly be NULL.

### CODE REVIEW

File: lib/ecl/tests/ecl_sum_restart.cpp [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+    ecl_sum_free(ecl_sum);
+
+    util_unlink("CASE1.SMSPEC");
+
+    verify_CASE2(num_dates);
+  }
+}
+
+int main() {
+  write_CASE2();
```
Comment (markusdregi) [inline lib/ecl/tests/ecl_sum_restart.cpp]: Should we consider running all the tests here ;)

File: lib/ecl/well_state.cpp [hunk 2]
```diff
@@ -649,7 +649,10 @@ time_t well_state_get_sim_time( const well_state_type * well_state ) {
    Will return NULL if no wellhead in this grid.
 */
 const well_conn_type * well_state_iget_wellhead( const well_state_type * well_state , int grid_nr) {
-  return well_state->index_wellhead[grid_nr];
```
Comment (markusdregi) [inline lib/ecl/well_state.cpp]: This change is already in master no?

**Comment by magnesj** [comment]
@joakim-hove : The current name of this PR is misleading, can you please rename it?

File: lib/ecl/tests/ecl_sum_restart.cpp (hunk 1 above)
Comment (joakim-hove) [reply lib/ecl/tests/ecl_sum_restart.cpp]: It is recursive - case2 calls case1

File: lib/ecl/well_state.cpp (hunk 2 above)
Comment (joakim-hove) [reply lib/ecl/well_state.cpp]: Think so - will do a rebase later

**Pull request merged**